mqtt/mqtt-bridge/src/client.rs

```diff
@@ -427,10 +427,10 @@ impl<H> MqttClientExt for MqttClient<H> {
 #[cfg(test)]
 /// Implements `MqttClientExt` for tests.
 impl<H> MqttClientExt for MqttClient<H> {
-    type PublishHandle = MockPublishHandle;
+    type PublishHandle = InFlightPublishHandle<MockallPublishHandleWrapper>;
 
-    fn publish_handle(&self) -> Result<Self::PublishHandle, ClientError> {
-        Ok(MockPublishHandle::new())
+    fn publish_handle(&self) -> Self::PublishHandle {
+        InFlightPublishHandle::new(MockallPublishHandleWrapper::new(), 100)
```

Comment:
Since the tests that use this don't test for in flight, I am making the in-flight very large so it will never block. We could avoid hardcoding this if we store a semaphore and create the publish handle each time. However, Denis thought it was cleaner if we create the publish handle once when creating the client.